packages/view/src/Charcoal/View/ViewAggregator.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Charcoal\View;
+
+use InvalidArgumentException;
+
+class ViewAggregator extends AbstractView
+{
+    /**
+     *  @var array<string, EngineInterface>
+     */
+    protected array $engines = [];
+
+    protected EngineInterface $engine;
+
+    /**
+     *  @var callable
+     */
+    protected $renderTemplateStringDecider;
+
+    /**
+     *  @var callable
+     */
+    protected $renderTemplateFileDecider;
+
+    /**
+     * @param array $data
+     */
+    public function __construct($data)
+    {
+        parent::__construct($data);
+
+        $this->setEngines($data['engines']);
+        $this->renderTemplateFileDecider = $data['file_decider'];
+        $this->renderTemplateStringDecider = $data['string_decider'];
```

Comment:
Hrm. Since the deciders are about selecting the appropriate engine, should we include "engine" in the names, or would that be too verbose?

For example:

```suggestion
        $this->renderTemplateFileEngineDecider = $data['file_engine_decider'];
        $this->renderTemplateStringEngineDecider = $data['string_engine_decider'];
```